packages/cli/src/pull/command.ts

```diff
@@ -0,0 +1,29 @@
+import yargs from 'yargs';
+import { build, ensure } from '../util/command-builders';
+import { Opts } from '../options';
+import * as o from '../options';
+
+export type DeployOptions = Required<
+  Pick<
+    Opts,
+    | 'command'
+    | 'log'
+    | 'logJson'
+    | 'statePath'
+    | 'projectPath'
+    | 'configPath'
+  >
+>;
+
+const options = [o.statePath, o.projectPath, o.configPath];
```

Comment:
The `projectPath` option doesn't seem quite right - the deploy config takes a `specPath` key, and when we write our output we write to `specPath`, not `projectPath`. There's a confusing mismatch there.

I may come back to you with more comments about the API, but at the very least `--project-path` needs to be `--spec-path`. You can keep `-p` and `--project-path` as aliases I guess, since `-s` is taken (for the record I don't think we should need to take -s as an argument at all, the CLI should manage that stuff automagically without user intervention)

Also, are all of these commands optional? Are they all required? My instinct is that the only one you really need is the config path (because you use a config for the first deploy, and afterwards all you do is sync the state of that config).